subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/internal/BuildScriptBuilder.java

```diff
@@ -747,6 +747,34 @@ public void writeCodeTo(PrettyPrinter printer) {
         }
     }
 
+    private static class GradleApiDepSpec extends AbstractStatement {
```

Comment:
We don't need to add gradleApi or gradleTestKit for Gradle plugin projects.

We have a "test source sets" feature in the JavaGradlePluginPlugin that automatically adds gradleApi and gradleTestKit as dependencies when a source set is added as a test. 

We also don't need to add a dependency on the project because I think we inject that in the JavaGradlePluginPlugin too.

You can check that this works because we don't do these things in the non-incubating version.

Reply:
I think I was trying to get this working prior to noticing the testSourceSets line in the example, so I made it work before recreating that line and making the prior work redundant.  I'll remove this.